src/data/ACTIONS/index.js

```diff
@@ -57,13 +60,25 @@ const ACTIONS = {
 	...RDM,
 }
 
+const addDefaultValues = obj => {
+	Object.keys(obj).forEach(key => {
+		const action = obj[key]
+		if (action.onGcd) {
+			action.castTime = action.castTime || DEFAULT_GCD_CASTTIME
+			action.cooldown = action.cooldown || DEFAULT_GCD_COOLDOWN
+		}
+	})
+	return obj
+}
+
 export const COOLDOWN_GROUPS = _.groupBy(ACTIONS, 'cooldownGroup')
 
 export const HIT_TYPES = {
 	CRIT: 2,
 }
 
 addExtraIndex(ACTIONS, 'id')
+addExtraIndex(addDefaultValues(ACTIONS), 'id')
```

Comment:
Feels hacky to me, but I can see _why_ you're doing it this way. Should be fine for now.

WoWA has an intermediary between the spell (action) data and the consumers via their `Spellbook` module, IIRC - I'll have a think about some alternative implementations for Later:tm:.

Reply:
We had a talk about this a few weeks back, just fyi.

```
αckwell08/05/2018
i can see the sanity behind it honestly
saves filling in data and saves keeping on toes with defaulting
take a look at data/STATUSES/index```

Though if the implementation's not how you imagined, I can fix that.

Reply:
Hahahaha, that's my memory for you.

This implementation is fine for now - that comment is more re: future development.